wallet/src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -180,16 +179,16 @@ sendMoney submitTx passphrase moneySource dstDistr policy = do
 
     logDebug "sendMoney: processed addrs"
 
-    let metasAndAdrresses = M.fromList $ zip (toList srcAddrs) (toList addrMetas)
+    let metasAndAddresses = M.fromList $ zip (toList srcAddrs) (toList addrMetas)
     allSecrets <- getSecretKeys
 
-    let getSigner addr = runIdentity $ do
-          let addrMeta =
-                  fromMaybe (error "Corresponding adress meta not found")
-                            (M.lookup addr metasAndAdrresses)
-          case runExcept $ getSKByAddressPure allSecrets (ShouldCheckPassphrase False) passphrase addrMeta of
-              Left err -> impureThrow err
-              Right sk -> withSafeSignerUnsafe sk (pure passphrase) pure
+    let
+        getSigner :: Address -> Maybe SafeSigner
+        getSigner addr = do
+          addrMeta <- M.lookup addr metasAndAddresses
+          sk <- rightToMaybe . runExcept $
```

Comment:
`runExcept` is probably 100% redundant here, but it's not you who added it and it's minor and will be refactored later anyway.